File: Java/String-2/catDog.java

public class string2 {
	
	// return true if the string "cat" and "dog" appear the same number of times in the given string
	public boolean catDog(String str) {
		
		boolean b = true;
		int cat = 0;
		int dog = 0;
		
		if (str.length() > 2) {
  		for (int i = 0; i < str.length()-2; i++) {
  			if (str.charAt(i) == 'c' && str.substring(i, i+3).equals("cat")) {
  				cat++;
  			}
  		}
  		
  		for (int i = 0; i < str.length()-2; i++) {
  			if (str.charAt(i) == 'd' && str.substring(i, i+3).equals("dog")) {
  				dog++;
  			}
  		}
  		
  		if (cat != dog) {
  			b = false;
  		}
		}
		
		return b;
		
	}

	public static void main(String[] args) {

		string2 o = new string2();
		
		System.out.println(o.catDog("catdog"));
		System.out.println(o.catDog("catcat"));
		System.out.println(o.catDog("1cat1cadodog"));

	}

}

File: JavaScript/String-2/catDog.js

// return true if the string 'cat' and 'dog' appear the same number of times in the given string
const catDog = (str) => {
	let cat = 0;
	let dog = 0;

	for (let i = 0; i < str.length - 2; i++) {
		if (str.substring(i, i + 3) === 'dog') {
			dog++;
		}

		if (str.substring(i, i + 3) === 'cat') {
			cat++;
		}
	}

	return cat === dog;
};

console.log(catDog('catdog'));
console.log(catDog('catcat'));
console.log(catDog('1cat1cadodog'));
